packages/@n8n/nodes-langchain/nodes/document_loaders/DocumentDefaultDataLoader/DocumentDefaultDataLoader.node.ts

```diff
@@ -20,13 +22,31 @@ import 'mammoth'; // for docx
 import 'epub2'; // for epub
 import 'pdf-parse'; // for pdf
 
+function getInputs(parameters: IDataObject) {
+	const inputs: INodeInputConfiguration[] = [];
+
+	const textSplittingMode = parameters?.textSplittingMode;
+	// If text splitting mode is 'custom' or does not exist (v1), we need to add an input for the text splitter
+	if (!textSplittingMode || textSplittingMode === 'custom') {
+		inputs.push({
+			displayName: 'Text Splitter',
+			maxConnections: 1,
+			type: 'ai_textSplitter',
```

Comment:
Downside of defining this as a function and calling toString() on it instead of inlining it is that you can't use enum values here, like you discovered.

This is why many of our nodes seem to just inline these,

https://github.com/n8n-io/n8n/pull/15915/files#diff-76afc25d81dbfea47a75d0ae4b1555db59b162ade6c650e6ff9baabb320a8ae6R53-R63

with that trick you can still use the enum values, as long as you pass those to the template literal.

But we're unlikely to ever change these enum values and your solution has the benefit of having the function typed so in my opinion we can keep this too, we seem to follow this pattern at ChainLLM node for instance too. 👍